### PR TITLE
`Development`: Fix exam exercise creation & e2e tests

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
There is a bug in the exam exercise creation & some failing e2e tests coming through #7875.
This PR reverts the wrong validation & fixes the e2e tests.

(it was not possible to create exam exercises, since the validation marked the creation form as invalid if there has been an unset exampleSolution Date. But Exams do not have dates.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Exam

1. Log in to Artemis
2. Navigate to Exam Administration
3. Create an exercise group and 1 exercise within
4. Verify that it is possible to create an exercise

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

## Summary by CodeRabbit

- **New Features**
	- Enhanced the exercise service logic for better validation of example solution publication dates.
- **Refactor**
	- Updated the handling of optional fields in the TitleChannelName component for improved form validation.
- **Tests**
	- Modified Cypress tests for exam and text exercise management to include better scroll behavior during test execution.
	- Added a new test case to verify date error handling in the TextExercise Management Update Component.

### CODE REVIEW

File: src/main/webapp/app/shared/form/title-channel-name/title-channel-name.component.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class TitleChannelNameComponent implements AfterViewInit, OnDestroy, OnIn
     @Input() initChannelName = true;
 
     @ViewChild('field_title') field_title: NgModel;
-    @ViewChild('field_channel_name') field_channel_name: NgModel;
+    @ViewChild('field_channel_name') field_channel_name?: NgModel;
```
Comment (milljoniaer) [inline src/main/webapp/app/shared/form/title-channel-name/title-channel-name.component.ts:22]: if `hideChannel` is true, this field is undefined.

File: src/main/webapp/app/exercises/shared/exercise/exercise.service.ts [hunk 2]
```diff
@@ -122,7 +122,6 @@ export class ExerciseService {
     hasExampleSolutionPublicationDateError(exercise: Exercise) {
         if (exercise.exampleSolutionPublicationDate) {
             return (
-                !dayjs(exercise.exampleSolutionPublicationDate).isValid() ||
```
Comment (milljoniaer) [inline src/main/webapp/app/exercises/shared/exercise/exercise.service.ts:125]: this line is generally a nice validation, but should be done not only for this date and should not be done for exams.
This PR reverts this wrong validation. I will do a followup PR to readd date validation in a proper way.

**Review by reschandreas — APPROVED** [review]
tested locally, works as expected. code lgtm. Thanks for fixing this!

**Comment by milljoniaer** [comment]
Fyi, the e2e tests did run through on bamboo, there seems to be a Github bug not showing the result correctly

**Review by farisd16 — APPROVED** [review]
tested on ts6, works as expected

**Review by mateusmm01 — APPROVED** [review]
Code lgtm

**Review by laurenzfb — APPROVED** [review]
tested on ts6, works

**Pull request merged**